src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -730,7 +730,9 @@ private void ExecutePutSubscriptionBatch(ClusterOperationContext context, Blitta
 
             PutSubscriptionCommand updateCommand = null;
             Exception exception = null;
-            var actionsByDatabase = new Dictionary<string, List<Func<Task>>>();
+            var actions = new List<Func<Task>>();
+            var databases = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Just use a dictionary? You have _one_ item per database, no?

Reply:
`ExecuteManyOnDispose` is expecting to get a `List<Func<Task>>`.
From a dictionary, I can get the values as `ValueCollection` and then need to call `ToList()`.

Reply:
@garayx Grisha is off, if you want to introduce changes, please handle them at this point via separate PR